src/FailureException.php

```diff
@@ -2,5 +2,5 @@
 
 namespace Amp\Mysql;
 
-class FailureException extends \Exception {
+class FailureException extends \Amp\Sql\FailureException {
 }
```

Comment:
Do we need the extension here and can't use the base class directly?

Reply:
Yes, we do. To just catch any SQL errors easily.

Reply:
For any SQL error it's fine to use `Amp\Sql\FailureException`, no?